custom_components/tuya_local/devices/ferroli_titano_twin.yaml

```diff
@@ -41,24 +32,68 @@ primary_entity:
       mapping:
         - step: 5
       unit: C
-      step: 5
     - id: 10
       type: integer
       name: current_temperature
-    - id: 101
-      type: integer
-      name: fault_code
-    - id: 102
-      type: string
-      name: workstate
-    - id: 103
-      type: boolean
-      name: antifreeze
-      mapping:
-        - dps_val: false
-          value: "off"
-        - dps_val: true
-          value: "on"
-    - id: 104
-      type: integer
-      name: target_temperature
\ No newline at end of file
+secondary_entities:
+  - entity: binary_sensor
+    name: Anti-Legionella
+    category: diagnostic
+    dps:
+      - id: 4
+        name: sensor
+        type: boolean
+        mapping:
+          - dps_val: true
+            icon: "mdi:shield-cross-outline"
+          - dps_val: false
+            icon: "mdi:shield-remove-outline"
+  - entity: binary_sensor
+    class: problem
+    name: Fault
+    category: diagnostic
+    dps:
+      - id: 101
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 0
+            value: false
+          - value: true
+            icon: "mdi:water-boiler-alert"
+  - entity: sensor
+    name: Working status
+    class: enum
```

Comment:
Would be better as a binary_sensor with class "running", I think (mappings would then have to map to values true and false).  Unless there are more than the two statuses mentioned in mappings (but the HA requirements for enum sensors are that the list of possible values is fixed and known).